### ISSUE TITLE
Delete old SMS from phone periodically

### ISSUE THREAD

**Issue by GoogleCodeExporter** [miramon98-1/android-sms#7]
```
One of the problems with the G1 is having too many SMS saved up or in a 
single thread.

Thus, it would be great if there is a setting to adjust the conditions 
under which an SMS message will be deleted from the phone, and how often it 
should be done.  After all, now we have access to all of them through Gmail 
anyway.

For example:

1. Delete if the message is more than 15 days old.
2. Delete if the message is in a thread with more than 100 messages AND it 
is older than the 100th most-recent message.
3. Delete if the message is in a thread with more than 100 messages AND it 
is older than the 100th most-recent message AND it is more than 15 days 
old.

So basically it boils down to 4 new settings:
1. Maximum number of messages per thread 
2. Maximum age of messages
3. Whether 1 and 2 should be in an AND or an OR relationship (default to 
OR)
4. Frequency of the purge


```

Original issue reported on code.google.com by `aris.lan...@gmail.com` on 26 Jan 2009 at 5:34

**Comment by GoogleCodeExporter** [comment]
```
This is a good suggestion as it is hard to delete a bunch of messages (but not 
all) 
from a thread using the default Messaging app.

I'm not sure whether such a complex rule setting could be done in a nice UI, 
but I'll 
keep the idea in mind.

It's nice to have the backup program not modify anything, so maybe we could 
create a 
second app that somehow plugs into SMS Backup.
```

Original comment by `chstu...@gmail.com` on 26 Jan 2009 at 6:04
- Changed state: **Accepted**
- Added labels: **Priority-Low**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `chstu...@gmail.com` on 26 Jan 2009 at 6:18

**Comment by GoogleCodeExporter** [comment]
```
Yeah, that's a good point.  Having a backup program deleting stuff would 
definitely 
raise some eyebrows, even if it only offered as an option.  

I wonder if the API would allow us write access to the messaging portion of the 
OS.  
I have a feeling that something like that would require root access.  In that 
case, 
it would make even more sense for this to be a separate app.  Otherwise, people 
without the JesusFreke mod would go bananas and claim that the purging function 
does 
not work.
```

Original comment by `aris.lan...@gmail.com` on 26 Jan 2009 at 6:33

**Comment by GoogleCodeExporter** [comment]
```
The API does provide means to edit and remove SMS, so that's not a problem.
```

Original comment by `chstu...@gmail.com` on 26 Jan 2009 at 6:38

**Comment by GoogleCodeExporter** [comment]
```
Ideally, I was hoping for an app that seamlessly "replaced" the Messaging app 
and put
all SMS/MMS online, with nice browsing / viewing from thither.  This "backup to
online" is great, but feels a bit like a "half-complete" solution, to me.  But 
that
may just be that I'm looking for a slightly different app than what this is 
supposed
to be.
```

Original comment by `john...@gmail.com` on 20 Oct 2009 at 5:48

**Comment by GoogleCodeExporter** [comment]
```
@chstuder: Instead of the complex rules, why not choose something like:

- Delete SMS (from phone) after (successful) backup (to email)

Also @johnsca: You can use Google Voice, which does exactly that. Turn off SMS 
forwarding and all your email is in Google Voice. However, there's a major flaw 
with 
the GV solution and that is, Data coverage is not as wide spread as the plain 
ol voice 
coverage (SMS travels in that garbage/throw away space of cell service).  The 
only 
solution to this is to keep getting SMS forwarded to your phone as well as get 
them 
through GV, but turn off the SMS app notification. That way, you get a dual sms 
feed, 
but only get notified for the GV SMS.
```

Original comment by `salah%ah...@gtempaccount.com` on 4 Dec 2009 at 9:56

**Comment by GoogleCodeExporter** [comment]
```
Great suggestion and very needed.

Old Treo's had an option to purge all SMS older than 2 weeks, 1 month etc.

Android lacks a way to purge old messages and keep newer ones which is really 
quite
essential to free up memory...

```

Original comment by `careyz...@gmail.com` on 18 Mar 2010 at 4:03

**Comment by GoogleCodeExporter** [comment]
```
well i agree with junoonie , the app should delete the old messages 
simultaneously or later after a successful back up because i have like 10,000 
or more sms on my phone at this very moment and it would be a tedious job to 
delete the messages after the backup. so if the app could do it for me in a 
faster and a better manner that would be really nice .
```

Original comment by `parth.go...@gmail.com` on 10 Aug 2010 at 8:40

**Comment by GoogleCodeExporter** [comment]
```
I will throw in my 2 cents on this as well.  I would LOVE to have the program 
be able to be configured in such a way that it backs up all messages over 30 
days old (or some user-configurable number), and once it's backed up, delete 
them from the phone.

I stumbled upon this app when looking for an SMS Archiving program.  This app 
seems to be right at the edge of that, and with a quick push could easily be 
that app...

GMail has given us a ton of storage space, while our phones do not.  I'd like 
to see this app run once a day (like in the middle of the night when I'm on 
wifi), archive all messages over 30 days old, then wipe the archived messages 
from my phone.  Thereby keeping the phone fairly clean, but still having access 
to the most recent SMS messages on the phone itself.
```

Original comment by `ETMegab...@gmail.com` on 12 Mar 2011 at 9:12
